components/collective-page/sections/Projects.tsx

```diff
@@ -0,0 +1,181 @@
+
+import { API_V2_CONTEXT } from '../../../lib/graphql/helpers';
+import useDebounced from '../../../lib/hooks/useDebounced';
+
+import { CONTRIBUTE_CARD_WIDTH } from '../../contribute-cards/constants';
+import ContributeProject from '../../contribute-cards/ContributeProject';
+import CreateNew from '../../contribute-cards/CreateNew';
+import { Box } from '../../Grid';
+import HorizontalScroller from '../../HorizontalScroller';
+import Link from '../../Link';
+import StyledButton from '../../StyledButton';
+import { P } from '../../Text';
+import { Input } from '../../ui/Input';
+import ContainerSectionContent from '../ContainerSectionContent';
+import ContributeCardsContainer from '../ContributeCardsContainer';
+import SectionTitle from '../SectionTitle';
+
+const ProjectSectionCardFields = gql`
+  fragment ProjectSectionCardFields on Project {
+    id
+    legacyId
+    slug
+    name
+    description
+    imageUrl
+    isActive
+    isArchived
+    backgroundImageUrl(height: 208)
+  }
+`;
+
+const CONTRIBUTE_CARD_PADDING_X = [15, 18];
+
+const ContributeCardContainer = styled(Box).attrs({ px: CONTRIBUTE_CARD_PADDING_X })(
+  css({
+    scrollSnapAlign: ['center', null, 'start'],
+  }),
+);
+
+type ProjectsProps = {
+  collective: {
+    slug: string;
+    name: string;
+    currency: string;
+    isActive: string;
+  };
+  projects: { id: number; isArchived?: boolean; isActive?: boolean; contributors?: object[] }[];
+  isAdmin: boolean;
+  showTitle: boolean;
+};
+
+function getContributeCardsScrollDistance(width) {
+  const oneCardScrollDistance = CONTRIBUTE_CARD_WIDTH + CONTRIBUTE_CARD_PADDING_X[0] * 2;
+  if (width <= oneCardScrollDistance * 2) {
+    return oneCardScrollDistance;
+  } else if (width <= oneCardScrollDistance * 4) {
+    return oneCardScrollDistance * 2;
+  } else {
+    return oneCardScrollDistance * 3;
+  }
+}
+
+export default function Projects(props: ProjectsProps) {
+  const { collective, isAdmin } = props;
+  const intl = useIntl();
+  const hasProjectsSection = (props.projects.length >= 0 && !collective.isActive) || isAdmin;
+
+  const [searchTerm, setSearchTerm] = React.useState('');
+  const deboucedSearchTerm = useDebounced(searchTerm, 1000);
+
+  const query = useQuery(
+    gql`
+      query ProjectsSection($slug: String, $searchTerm: String) {
+        account(slug: $slug) {
+          ... on AccountWithProjects {
+            projects(offset: 0, limit: 10, term: $searchTerm) {
+              totalCount
+              nodes {
+                ...ProjectSectionCardFields
+              }
+            }
+          }
+        }
+      }
+
+      ${ProjectSectionCardFields}
+    `,
+    {
+      context: API_V2_CONTEXT,
+      variables: {
+        slug: props.collective.slug,
+        searchTerm: deboucedSearchTerm,
+      },
+      skip: isEmpty(deboucedSearchTerm),
+    },
+  );
+
+  const refetch = query.refetch;
+  React.useEffect(() => {
+    refetch();
+  }, [deboucedSearchTerm, refetch]);
```

Comment:
I don't get why we need to manually refetch. Since it's not a `lazyQuery` the refetch should happen automatically when variables change, right?